From 8d768437c8842d963426dcb52b32f5640e4c5df7 Mon Sep 17 00:00:00 2001
From: "Philip, Avinash" <avinashphilip@ti.com>
Date: Fri, 9 Mar 2012 22:03:22 +0530
Subject: [PATCH 0878/1092] ARM: OMAP2+: elm: Creates device on available
 profiles.

This patch corrects creation of ELM device for profiles where NAND is
available. Previously ELM device was created from arch_initcal().

Signed-off-by: Philip, Avinash <avinashphilip@ti.com>
---
 arch/arm/mach-omap2/board-am335xevm.c |  1 +
 arch/arm/mach-omap2/devices.c         | 10 +---------
 arch/arm/mach-omap2/devices.h         |  1 +
 3 files changed, 3 insertions(+), 9 deletions(-)

diff --git a/arch/arm/mach-omap2/board-am335xevm.c b/arch/arm/mach-omap2/board-am335xevm.c
index 96cfe07..5dec7aa 100644
--- a/arch/arm/mach-omap2/board-am335xevm.c
+++ b/arch/arm/mach-omap2/board-am335xevm.c
@@ -1184,6 +1184,7 @@ static void evm_nand_init(int evm_id, int profile)
 	gpmc_device[0].flag = GPMC_DEVICE_NAND;
 
 	omap_init_gpmc(gpmc_device, sizeof(gpmc_device));
+	omap_init_elm();
 }
 
 /* TPS65217 voltage regulator support */
diff --git a/arch/arm/mach-omap2/devices.c b/arch/arm/mach-omap2/devices.c
index c89e11d..33ecedf 100644
--- a/arch/arm/mach-omap2/devices.c
+++ b/arch/arm/mach-omap2/devices.c
@@ -524,9 +524,7 @@ static void omap_init_mcspi(void)
 static inline void omap_init_mcspi(void) {}
 #endif
 
-#ifdef CONFIG_SOC_OMAPAM33XX
-
-static int __init omap_init_elm(void)
+int __init omap_init_elm(void)
 {
 	int id = -1;
 	struct platform_device *pdev;
@@ -551,11 +549,6 @@ static int __init omap_init_elm(void)
 	return 0;
 }
 
-#else
-static inline  int omap_init_elm(void) { }
-#endif
-
-
 static struct resource omap2_pmu_resource = {
 	.start	= 3,
 	.end	= 3,
@@ -1351,7 +1344,6 @@ static int __init omap2_init_devices(void)
 	omap_init_camera();
 	omap_init_mbox();
 	omap_init_mcspi();
-	omap_init_elm();
 	omap_init_pmu();
 	omap_hdq_init();
 	omap_init_sti();
diff --git a/arch/arm/mach-omap2/devices.h b/arch/arm/mach-omap2/devices.h
index 8174cf3..4164d78 100644
--- a/arch/arm/mach-omap2/devices.h
+++ b/arch/arm/mach-omap2/devices.h
@@ -19,5 +19,6 @@ int omap3_init_camera(struct isp_platform_data *pdata);
 int __init am335x_register_mcasp(struct snd_platform_data *pdata, int ctrl_nr);
 extern int __init am33xx_register_tsc(struct tsc_data *pdata);
 extern void register_ehrpwm(int max_freq);
+extern int __init omap_init_elm(void);
 
 #endif
-- 
1.7.11.2

